lib/issues_and_messages_controller_patch.rb

```diff
@@ -8,5 +8,49 @@ module InstanceMethods
       def init_users_cache
         @users_cache = Hash.new
       end
+
+      def expand_collapse_notes_or_messages(entities, getPreview, shouldExcludeEntity, getRegExToHideContent, getRegExesToHideOtherElements) 
+        if (AdvancedMessengerSettings.get_setting(:disable_fix_for_scroll_to_anchor) == '1' || !User.current.logged?)
+          return
+        end
+
+        entities.each do |entity|
+          if shouldExcludeEntity.call(entity)
+            next
+          end  
+          
+          read_by_current_user = JSON.parse(entity.read_by_users)[User.current.id.to_s]
+          expanded = false
+          if read_by_current_user == nil || read_by_current_user["read"] == 1 || read_by_current_user["read"] == nil && !read_by_current_user["collapsed"]
+            expanded = true
+          end 
+          
+          #==========Add the content preview besides the actual content and hide/show the content/content preview==============
+          truncated_notes = escape_javascript(getPreview.call(entity))
```

Comment:
no that problem was related to a mistake that I had made then: I called 
`substr(escape_javascript(content), 100)` instead of `escape_javascript(substr(content, 100))`